extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -504,24 +504,34 @@ export class AmpConsent extends AMP.BaseElement {
       ) {
         this.updateCacheIfNotNull_(
           response['consentStateValue'],
-          response['consentString'] || undefined
+          response['consentString'] || undefined,
+          response['gdprApplies'] !== undefined
+            ? response['gdprApplies']
+            : !!response['consentRequired']
         );
       }
     });
   }
 
   /**
    * Sync with local storage if consentRequired is true.
+   * Treat all three values as a grouping, only update cache
+   * if responseStateValue is not null
    * @param {string=} responseStateValue
    * @param {string=} responseConsentString
+   * @param {boolean=} responseGdprApplies
```

Comment:
If we use `consetRequired` value, then `responseGdprApplies` would never be undefined. Can we assume it's always a boolean value?

Reply:
Yup